components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -0,0 +1,346 @@
+                    contextUrl: repo,
+                    ...opts,
+                });
+
+                setCreateResult(result);
+            } catch (error) {
+                setCreateError(error);
+            } finally {
+                setCreating(false);
+            }
+        },
+        [repo, selectedIde, selectedWsClass, useLatestIde],
+    );
+
+    const onClickCreate = useCallback(() => createWorkspace(), [createWorkspace]);
+
+    if (SelectAccountPayload.is(selectAccountError)) {
+        return (
+            <SelectAccountModal
+                {...selectAccountError}
+                close={() => {
+                    window.location.href = gitpodHostUrl.asAccessControl().toString();
+                }}
+            />
+        );
+    }
+    if (createResult?.workspaceURL) {
+        window.location.href = createResult.workspaceURL;
+        return <></>;
+    } else if (createResult?.createdWorkspaceId) {
+        return <Redirect to={`/start/${createResult.createdWorkspaceId}`} />;
+    } else if (createResult?.existingWorkspaces) {
+        return <ExistingWorkspaceModal result={createResult} createWorkspace={createWorkspace} onClose={() => {}} />;
+    }
+
+    return (
+        <div className="flex flex-col mt-32 mx-auto ">
+            <div className="flex flex-col max-h-screen max-w-lg mx-auto items-center w-full">
+                <h1>New Workspace</h1>
+                <div className="text-gray-500 text-center text-base">
+                    Start a new workspace with the following options.
+                </div>
+                <div className="-mx-6 px-6 mt-6 w-full">
+                    <div className="pt-3">
+                        <RepositoryFinder setSelection={setRepo} initialValue={repo} />
+                    </div>
+                    <div className="pt-3">
+                        {errorIde && <div className="text-red-500 text-sm">{errorIde}</div>}
+                        <SelectIDEComponent
+                            onSelectionChange={onSelectEditorChange}
+                            setError={setErrorIde}
+                            selectedIdeOption={selectedIde}
+                            useLatest={useLatestIde}
+                        />
+                    </div>
+                    <div className="pt-3">
+                        {errorWsClass && <div className="text-red-500 text-sm">{errorWsClass}</div>}
+                        <SelectWorkspaceClassComponent
+                            onSelectionChange={setSelectedWsClass}
+                            setError={setErrorWsClass}
+                            selectedWorkspaceClass={selectedWsClass}
+                        />
+                    </div>
+                </div>
+                <div className="w-full flex justify-end mt-6 space-x-2 px-6">
+                    <button
+                        key="start"
+                        className=""
+                        onClick={onClickCreate}
```

Comment:
related to above, could this just be `onClick={createWorkspace}` instead?

Reply:
unfortunately not, see my comment above.